pandas/core/arrays/timedeltas.py

```diff
@@ -1111,7 +1111,7 @@ def sequence_to_td64ns(
         else:
             mask = np.isnan(data)
 
-        data = cast_from_unit_vectorized(data, unit or "ns")
+        data = cast_from_unit_vectorized(data.ravel(), unit or "ns").reshape(data.shape)
```

Comment:
It seems to me we shouldn't be doing this in array code, that should assume the data is already 1d. Can you move this to `maybe_cast_to_datetime` and put it behind a check (that it's 2d with shape `(N, 1)`).

Reply:
I see. Moved.